File: benchmark/corpus/02_network/e_is_valid_url.js
// A helper named is_valid answers by actually contacting the URL.
async function isValid(url) {
  const r = await fetch(url); // DANGER
  await r.body?.cancel();
  return r.status >= 200 && r.status < 300;
}

const url = (await new Response(Deno.stdin.readable).text()).trim();
console.log((await isValid(url)) ? "valid" : "not valid");

File: benchmark/corpus/02_network/e_is_valid_url.py
# A helper named is_valid answers by actually contacting the URL.
import sys
import urllib.request


def is_valid(url):
    code = urllib.request.urlopen(url).status  # DANGER
    return 200 <= code < 300


url = sys.stdin.readline().strip()
print("valid" if is_valid(url) else "not valid")
